src/abstractdataframe/selection.jl

```diff
@@ -170,56 +170,208 @@ function normalize_selection(idx::AbstractIndex,
     return (wanttable ? AsTable(c) : c) => fun => newcol
 end
 
-function select_transform!(nc::Pair{<:Union{Int, AbstractVector{Int}, AsTable},
-                                    <:Pair{<:Base.Callable, Symbol}},
+function _transformation_helper(df::AbstractDataFrame, col_idx, @nospecialize(fun))
+    if col_idx === nothing
+        return fun(df)
```

Comment:
Okay so we are keeping this in transform? I voted against [here](https://github.com/JuliaData/DataFrames.jl/issues/2410#issuecomment-695823805) but I don't know if there was ever a consensus. 

It's fine to add! But just to confirm.

Reply:
Also, the `do` syntax here is really nice, I didnt realize that was added. I approve this.

Reply:
We allow `select(fun, df)` and `select(df, fun)` but not allow `select(fun => AsTable, df)` and `select(df, fun => AsTable)` (or anything else in place of `AsTable`). In short only a stand-alone `::Callable` is allowed as a special case.

Reply:
Thanks. That takes care of my 2nd comment, but I think my first one (on the main discussion) still applies.